packages/schema/parse/src/comparison/comparers.ts

```diff
@@ -0,0 +1,903 @@
+import {
+  QueryDefinition,
+  EnumDefinition,
+  ObjectDefinition,
+  PropertyDefinition,
+  GenericDefinition,
+  ArrayDefinition,
+  MethodDefinition,
+  ImportedQueryDefinition,
+  ImportedEnumDefinition,
+  ImportedObjectDefinition,
+  ImportedDefinition,
+} from "../typeInfo/definitions";
+import { TypeInfo } from "../typeInfo";
+import { compareSets, SetComparisonType } from "./compareSets";
+
+export type ImportDefinition = { type: string };
+
+export enum VersionRelease {
+  PATCH,
+  MINOR,
+  MAJOR,
+}
+
+export interface CompareOptions {
+  shortCircuit?: VersionRelease;
+}
+
+export interface CompareResult {
+  versionRelease: VersionRelease;
+  hasShortCircuit: boolean;
+}
+
+export function compare(
+  t1: TypeInfo,
+  t2: TypeInfo,
+  options?: CompareOptions
+): CompareResult {
+  const result: CompareResult = {
+    versionRelease: VersionRelease.PATCH,
+    hasShortCircuit: false,
+  };
+
+  const enumResult = compareEnumTypes(t1.enumTypes, t2.enumTypes, options);
+  if (enumResult.hasShortCircuit) return enumResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    enumResult.versionRelease
+  );
+
+  const objectResult = compareObjectTypes(
+    t1.objectTypes,
+    t2.objectTypes,
+    options
+  );
+  if (objectResult.hasShortCircuit) return objectResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    objectResult.versionRelease
+  );
+
+  const queryResult = compareQueryTypes(t1.queryTypes, t2.queryTypes, options);
+  if (queryResult.hasShortCircuit) return queryResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    queryResult.versionRelease
+  );
+
+  const importedEnumResult = compareImportedEnumTypes(
+    t1.importedEnumTypes,
+    t2.importedEnumTypes,
+    options
+  );
+  if (importedEnumResult.hasShortCircuit) return importedEnumResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    importedEnumResult.versionRelease
+  );
+
+  const importedObjectResult = compareImportedObjectTypes(
+    t1.importedObjectTypes,
+    t2.importedObjectTypes,
+    options
+  );
+  if (importedObjectResult.hasShortCircuit) return importedObjectResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    importedObjectResult.versionRelease
+  );
+
+  const importedQueryResult = compareImportedQueryTypes(
+    t1.importedQueryTypes,
+    t2.importedQueryTypes,
+    options
+  );
+  if (importedQueryResult.hasShortCircuit) return importedQueryResult;
+  result.versionRelease = Math.max(
+    result.versionRelease,
+    importedQueryResult.versionRelease
+  );
+
+  return result;
+}
+
+export function compareGenericType(
```

Comment:
It'd be a lot more readable if each of these was it's own file. It's generally a really good practice to keep files short. It's easier to find the logic you need from the file explorer and one can use that as a high level overview of the logic.
Something like:

src/comparison/VersionRelease.ts
src/comparison/compare.ts
src/comparison/comparers/compareGenericType.ts
src/comparison/comparers/compareQueryType.ts
...